### PR TITLE
Add sensuctl command

### LINKED ISSUE #1923: Document the "command" subcommand for sensuctl
Documentation is required for the new `sensuctl command` subcommand implemented in https://github.com/sensu/sensu-go/pull/3274.

* `sensuctl command install [ALIAS] ([NAME]:[VERSION] | --url [ARCHIVE_URL] --checksum [ARCHIVE_CHECKSUM]) [flags]`
  This command installs a sensuctl command from Bonsai or a URL.
  `ALIAS` is mandatory, and one of `NAME` (with an optional `:VERSION`) or `--url ARCHIVE_URL --checksum ARCHIVE_CHECKSUM` are required.
  `:VERSION` is optional, if not specified it will fetch the latest version from Bonsai.
* `sensuctl command exec [ALIAS] [args] [flags]`
  This command executes a sensuctl command plugin via it's asset's `bin/entrypoint` executable.
  `ALIAS` is mandatory.
  `args` is passed through to the `bin/entrypoint` executable.
* `sensuctl command list [flags]`
  This command lists installed sensuctl commands.
* `sensuctl command delete [ALIAS] [flags]`
  This command deletes sensuctl commands.

## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [x] New content (guide wanted)
- [ ] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Add sensuctl command info to sensuctl reference

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/1923

## Review Instructions

Outstanding questions to resolve:

- [x] Confirm my explanation of `[ALIAS]`: is it a unique name for the command that the user adds?
- [x] Confirm and complete my examples. I don't know if they are correct, and I know they are not complete.
- [x] Specify what to replace `[args]` with.
- [x] Add a brief explanation (1 or 2 sentences) of what command we mean. For example, we are basically talking about these, right? https://docs.sensu.io/sensu-go/5.14/reference/checks/#check-commands
- [x] Confirm this will be released in 5.15

### CODE REVIEW

File: content/sensu-go/5.15/sensuctl/reference.md [hunk 1]
```diff
@@ -798,14 +802,71 @@ fetching bonsai asset: sensu/sensu-slack-handler:1.0.3
...
+sensuctl command install [ALIAS] (sensu/sensu-email-handler:0.2.0) -tf
+...
+**NEEDED**
+...
+{{< /highlight >}}
+
+To install a command from a URL:
+
+{{< highlight shell >}}
+sensuctl command install [ALIAS] (--url https://gist.githubusercontent.com/amdprophet/43790e007e949aed19131b100e1576af/raw/0c3a1af85481db2bcfbf9f7be513fa7fc0d0fc18/asset.yml --checksum **NEEDED*)
```
Comment (amdprophet) [inline content/sensu-go/5.15/sensuctl/reference.md]: When installing a sensuctl command asset via a URL, the URL must point to an asset tarball (e.g. `https://path/to/asset.tar.gz`).

Asset definitions with multiple asset builds are currently only supported via Bonsai.

**Comment by amdprophet** [comment]
`[args]` for the most part are just any of the supported global flags (e.g. `--cache-dir`, `--namespace`, etc.).

```
# sensuctl command install -h
installs a sensuctl command from Bonsai or a URL

Usage:  sensuctl command install [ALIAS] ([NAME]:[VERSION] | --url [ARCHIVE_URL] --checksum [ARCHIVE_CHECKSUM]) [flags]

Flags:
      --checksum string   specifies the checksum (SHA512) of the command archive from the --url flag
  -h, --help              help for install
      --url string        specifies a URL to fetch a sensuctl command archive from

Global Flags:
      --api-url string             host URL of Sensu installation
      --cache-dir string           path to directory containing cache & temporary files (default "/Users/justin/Library/Caches/sensu/sensuctl")
      --config-dir string          path to directory containing configuration files (default "/Users/justin/.config/sensu/sensuctl")
      --insecure-skip-tls-verify   skip TLS certificate verification (not recommended!)
      --namespace string           namespace in which we perform actions (default "default")
      --trusted-ca-file string     TLS CA certificate bundle in PEM format
```

```
# sensuctl command exec -h
executes a sensuctl command plugin

Usage:  sensuctl command exec [ALIAS] [args] [flags]

Flags:
  -h, --help   help for exec

Global Flags:
      --api-url string             host URL of Sensu installation
      --cache-dir string           path to directory containing cache & temporary files (default "/Users/justin/Library/Caches/sensu/sensuctl")
      --config-dir string          path to directory containing configuration files (default "/Users/justin/.config/sensu/sensuctl")
      --insecure-skip-tls-verify   skip TLS certificate verification (not recommended!)
      --namespace string           namespace in which we perform actions (default "default")
      --trusted-ca-file string     TLS CA certificate bundle in PEM format
```

```
# sensuctl command list -h
lists installed sensuctl commands

Usage:  sensuctl command list [flags]

Flags:
  -h, --help   help for list

Global Flags:
      --api-url string             host URL of Sensu installation
      --cache-dir string           path to directory containing cache & temporary files (default "/Users/justin/Library/Caches/sensu/sensuctl")
      --config-dir string          path to directory containing configuration files (default "/Users/justin/.config/sensu/sensuctl")
      --insecure-skip-tls-verify   skip TLS certificate verification (not recommended!)
      --namespace string           namespace in which we perform actions (default "default")
      --trusted-ca-file string     TLS CA certificate bundle in PEM format
```

```
# sensuctl command delete -h
delete sensuctl commands

Usage:  sensuctl command delete [ALIAS] [flags]

Flags:
  -h, --help           help for delete
      --skip-confirm   skip interactive confirmation prompt

Global Flags:
      --api-url string             host URL of Sensu installation
      --cache-dir string           path to directory containing cache & temporary files (default "/Users/justin/Library/Caches/sensu/sensuctl")
      --config-dir string          path to directory containing configuration files (default "/Users/justin/.config/sensu/sensuctl")
      --insecure-skip-tls-verify   skip TLS certificate verification (not recommended!)
      --namespace string           namespace in which we perform actions (default "default")
      --trusted-ca-file string     TLS CA certificate bundle in PEM format
```

File: content/sensu-go/5.15/sensuctl/reference.md [hunk 2]
```diff
@@ -798,14 +802,71 @@ fetching bonsai asset: sensu/sensu-slack-handler:1.0.3
...
+Replace `[ALIAS]` with a unique name for the command. `[ALIAS]` is required.
+
+You must specify **either** `[NAME]` (with optional `[:VERSION]`) **or** `--url ARCHIVE_URL --checksum ARCHIVE_CHECKSUM`.
+
+Replace `[NAME]` with the name of the asset from Bonsai.
+`[:VERSION]` is only required if you require a specific version or are pinning to a specific version. If you do not specify a version, sensuctl will fetch the latest version from Bonsai.
+
+Replace `[flags]` with the resource-specific flags you want to use.
+
+For example, to install a command from the [Sensu Go Email Handler Plugin][35] with the [`-t` and -`f` flags][36] to retrieve the "To" and "From" email addresses:
```
Comment (amdprophet) [inline content/sensu-go/5.15/sensuctl/reference.md]: The flags here are just for the install command and will not be saved as part of the command.

File: content/sensu-go/5.15/sensuctl/reference.md [hunk 3]
```diff
@@ -798,14 +802,71 @@ fetching bonsai asset: sensu/sensu-slack-handler:1.0.3
...
+...
+{{< /highlight >}}
+
+#### Execute commands
+
+Use `sensuctl command exec [ALIAS] [args] [flags]` to execute a sensuctl command plugin via its asset's bin/entrypoint executable.
+
+Replace `[ALIAS]` with with a unique name for the command. `[ALIAS]` is required.
+
+Replace `[args]` with **NEEDED**. `[args]` is passed through to the bin/entrypoint executable.
```
Comment (amdprophet) [inline content/sensu-go/5.15/sensuctl/reference.md]: This is correct however we should mention that in order for flags to be passed to the `bin/entrypoint` executable, they must be specified after a double dash surrounded by spaces ` -- `.

For example:
```
sensuctl command exec mycommand arg1 arg2 --cache-dir /tmp -- --flag1 --flag2=value
```

The `--cache-dir` flag will be parsed by sensuctl, but all flags after the `--` will be parsed by `bin/entrypoint`. The full command being run by sensuctl exec in this case would be:

```
bin/entrypoint arg1 arg2 --flag1 --flag2=value
```

File: content/sensu-go/5.15/sensuctl/reference.md (hunk 1 above)
Comment (amdprophet) [reply content/sensu-go/5.15/sensuctl/reference.md]: A working example:

```
sensuctl command install command-test --url https://github.com/amdprophet/command-test/releases/download/v0.0.4/command-test_0.0.4_darwin_amd64.tar.gz --checksum 8b15a170e091dab42256fe64ca7c4a050ed49a9dbfd6c8129c95506a8a9a91f2762ac1a6d24f4fc545430613fd45abc91d3e5d3605fcfffb270dcf01996caa7f
```

**Comment by amdprophet** [comment]
For further explanation of sensuctl commands see: https://docs.google.com/document/d/1veQ30ONMDFfZOayZLpAjmzgr1V96ahUK62uWWnaQiU4

File: content/sensu-go/5.15/sensuctl/reference.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/5.15/sensuctl/reference.md]: Awesome, thank you!

**Comment by hillaryfraley** [comment]
Thank you, @amdprophet ! All updated.

**Comment by amdprophet** [comment]
These changes look good to me!

I forgot to mention that we'll want to document the environment variables that are passed to the command when using `command exec`:

```
SENSU_API_URL (url of the sensu backend api in sensuctl)
SENSU_NAMESPACE (name of the current namespace in sensuctl)
SENSU_FORMAT (set output format in sensuctl e.g. json, yaml, etc.)
SENSU_ACCESS_TOKEN (current api access token in sensuctl)
SENSU_ACCESS_TOKEN_EXPIRES_AT (timestamp specifying when the above access token expires)
SENSU_REFRESH_TOKEN (refresh token used to obtain a new access token)
SENSU_TRUSTED_CA_FILE (path to a trusted CA file if set in sensuctl)
SENSU_INSECURE_SKIP_TLS_VERIFY (boolean value that can be set to skip TLS verification)
```

I'll approve this after this addition and I apologize for not including it in the initial information.

**Comment by hillaryfraley** [comment]
Thank you, @amdprophet ! I added the descriptions in the Environment Variables section and linked to them from the sensuctl command exec section. It seemed to make more sense to define them in the Env Var section (we had a usage example there but not the descriptions you provided) -- wdyt?

Edit: seems like this should be part of the 5.15 release notes, is that correct? I saw that the original issue specified a changelog entry was needed, so I went ahead and added this as a new feature in the 5.15 release notes.

**Review by amdprophet — APPROVED** [review]
Looks great! Yes, this will be a part of 5.15.

**Pull request merged**